pkg/inject/sidecar.go

```diff
@@ -104,6 +104,8 @@ func container(jaeger *v1.Jaeger) corev1.Container {
 	jgCompactTrft := util.GetPort("--processor.jaeger-compact.server-host-port=", args, 6831)
 	jgBinaryTrft := util.GetPort("--processor.jaeger-binary.server-host-port=", args, 6832)
 
+	commonSpec := util.Merge([]v1.JaegerCommonSpec{jaeger.Spec.Agent.JaegerCommonSpec, jaeger.Spec.JaegerCommonSpec})
```

Comment:
What does this do? Why isn't it enough to use just `Spec.Agent.Resources`?

Reply:
It allows common resource requirements to be defined in `jaeger.Spec.Resources` and only agent specific requirements defined in `jaeger.Spec.Agent.Resources`. Its a pattern used for all parts of the common spec (e.g. volumnes, mounts, annotations, etc).

Reply:
Yes. It has been done from a consistency perspective - but if is not reasonable, then we just need to make sure it is documented clearly that for agent as a sidecar (or agent in general), the parent resources are not inherited.

Reply:
I would maybe prefer applying only resources on the agent spec as sidecar can populate to a lot of pods but I don't have strong objections.